### ISSUE TITLE
Can't generate types for stream-transform because of parsing error

### ISSUE THREAD

**Issue by jvican** [ScalablyTyped/Converter#478]
I was trying to generate types for [stream-transform](https://www.npmjs.com/package/stream-transform) version 3.2.0 and I'm currently getting the following error:

```
[error] 2022-09-24T16:56:08.767Z PhaseRes.scala:78  Caught exception: Error while parsing: /Users/jvican/Code/playground-scala/goodreads-to-zotero/node_modules/stream-transform/package.json: DecodingFailure(String, List(DownField(types))) [id => stream-transform, thread => 1, phase => typescript]java.lang.RuntimeException: Error while parsing: /Users/jvican/Code/playground-scala/goodreads-to-zotero/node_modules/stream-transform/package.json: DecodingFailure(String, List(DownField(types)))
	at scala.sys.package$.error(package.scala:30)
	at org.scalablytyped.converter.internal.Json$.force(Json.scala:103)
	at org.scalablytyped.converter.internal.Json$.force(Json.scala:99)
	at org.scalablytyped.converter.internal.Json$.opt(Json.scala:132)
	at org.scalablytyped.converter.internal.importer.Source$TsLibSource.packageJsonOpt(Source.scala:38)
	at org.scalablytyped.converter.internal.importer.Source$TsLibSource.packageJsonOpt$(Source.scala:37)
	at org.scalablytyped.converter.internal.importer.Source$FromFolder.packageJsonOpt$lzycompute(Source.scala:50)
	at org.scalablytyped.converter.internal.importer.Source$FromFolder.packageJsonOpt(Source.scala:50)
	at org.scalablytyped.converter.internal.importer.Source$.findShortenedFiles(Source.scala:89)
	at org.scalablytyped.converter.internal.importer.Source$TsLibSource.shortenedFiles(Source.scala:45)
	at org.scalablytyped.converter.internal.importer.Source$TsLibSource.shortenedFiles$(Source.scala:45)
	at org.scalablytyped.converter.internal.importer.Source$FromFolder.shortenedFiles$lzycompute(Source.scala:50)
	at org.scalablytyped.converter.internal.importer.Source$FromFolder.shortenedFiles(Source.scala:50)
	at org.scalablytyped.converter.internal.importer.PathsFromTsLibSource$.apply(PathsFromTsLibSource.scala:18)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.apply(Phase1ReadTypescript.scala:136)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.apply(Phase1ReadTypescript.scala:21)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$4(PhaseRunner.scala:71)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$3(PhaseRunner.scala:71)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$1(PhaseRunner.scala:70)
	at org.scalablytyped.converter.internal.phases.PhaseCache.getOrElse(PhaseCache.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$2(PhaseRunner.scala:47)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.immutable.NewRedBlackTree$._foreachKey(RedBlackTree.scala:291)
	at scala.collection.immutable.NewRedBlackTree$.foreachKey(RedBlackTree.scala:287)
	at scala.collection.immutable.TreeSet.foreach(TreeSet.scala:254)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.immutable.TreeSet.scala$collection$SetLike$$super$map(TreeSet.scala:134)
	at scala.collection.SetLike.map(SetLike.scala:105)
	at scala.collection.SetLike.map$(SetLike.scala:105)
	at scala.collection.immutable.TreeSet.map(TreeSet.scala:134)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$5(PhaseRunner.scala:61)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.$anonfun$apply$12(Phase1ReadTypescript.scala:130)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.$anonfun$apply$9(Phase1ReadTypescript.scala:125)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.$anonfun$apply$7(Phase1ReadTypescript.scala:114)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.$anonfun$apply$4(Phase1ReadTypescript.scala:109)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.$anonfun$apply$3(Phase1ReadTypescript.scala:107)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.apply(Phase1ReadTypescript.scala:79)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.apply(Phase1ReadTypescript.scala:21)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$4(PhaseRunner.scala:71)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$3(PhaseRunner.scala:71)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$1(PhaseRunner.scala:70)
	at org.scalablytyped.converter.internal.phases.PhaseCache.getOrElse(PhaseCache.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$2(PhaseRunner.scala:47)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.immutable.NewRedBlackTree$._foreachKey(RedBlackTree.scala:291)
	at scala.collection.immutable.NewRedBlackTree$._foreachKey(RedBlackTree.scala:290)
	at scala.collection.immutable.NewRedBlackTree$.foreachKey(RedBlackTree.scala:287)
	at scala.collection.immutable.TreeSet.foreach(TreeSet.scala:254)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.immutable.TreeSet.scala$collection$SetLike$$super$map(TreeSet.scala:134)
	at scala.collection.SetLike.map(SetLike.scala:105)
	at scala.collection.SetLike.map$(SetLike.scala:105)
	at scala.collection.immutable.TreeSet.map(TreeSet.scala:134)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$5(PhaseRunner.scala:61)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.apply(Phase1ReadTypescript.scala:166)
	at org.scalablytyped.converter.internal.importer.Phase1ReadTypescript.apply(Phase1ReadTypescript.scala:21)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$4(PhaseRunner.scala:71)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$3(PhaseRunner.scala:71)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap(PhaseRes.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRes.flatMap$(PhaseRes.scala:24)
	at org.scalablytyped.converter.internal.phases.PhaseRes$Ok.flatMap(PhaseRes.scala:33)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$1(PhaseRunner.scala:70)
	at org.scalablytyped.converter.internal.phases.PhaseCache.getOrElse(PhaseCache.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$1(PhaseRunner.scala:47)
	at org.scalablytyped.converter.internal.phases.PhaseCache.getOrElse(PhaseCache.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$1(PhaseRunner.scala:47)
	at org.scalablytyped.converter.internal.phases.PhaseCache.getOrElse(PhaseCache.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.$anonfun$doNext$1(PhaseRunner.scala:47)
	at org.scalablytyped.converter.internal.phases.PhaseCache.getOrElse(PhaseCache.scala:26)
	at org.scalablytyped.converter.internal.phases.PhaseRunner$.go(PhaseRunner.scala:47)
	at org.scalablytyped.converter.cli.Main$.$anonfun$main$12(Main.scala:289)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at org.scalablytyped.converter.cli.Main$.main(Main.scala:289)
	at org.scalablytyped.converter.cli.Main.main(Main.scala)
```

Looks like the contains a top-level `types` json key that scalably typed doesn't know about:

```
{
  "version": "2.1.3",
  "name": "stream-transform",
  ...
  "types": [
    "./lib/es5/index.d.ts",
    "./lib/es5/sync.d.ts",
    "./lib/index.d.ts",
    "./lib/sync.d.ts"
  ],
  "dependencies": {
    "mixme": "^0.5.1"
  },
  "gitHead": "0fd5209b6862655c384cda052abf38019b959e70"
}

```

P.S. Thanks for the project @oyvindberg 😄

**Comment by jvican** [comment]
Looks like other stream-transform versions also have other type-related keys they use, which we might want to support, see https://github.com/adaltas/node-csv/blob/master/packages/stream-transform/package.json#L86-L103

**Comment by oyvindberg** [comment]
Interesting, thanks! I'm working towards #474 and #463 now, and will solve this at the same time :)

> P.S. Thanks for the project @oyvindberg 😄

❤️

**Comment by oyvindberg** [comment]
Fixed in #474

**Comment by jvican** [comment]
Thanks!
